src/pages/index.js

```diff
@@ -3,141 +3,201 @@ import "../pages/index.css";
+    handleCardLike
   });
-  return card;
+  return card.getCard();
 }
 
 //IMAGE POPUP
 const imagePopup = new PopupWithImage("#preview-image-modal");
 imagePopup.setEventListeners();
 
-//FORM POPUP
+//DELETE CARD POPUP
+const deleteCardPopup = new PopupWithConfirmation("#card__delete-modal");
+deleteCardPopup.setEventListeners();
+
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setSubmitText(true, "Deleting...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        deleteCardPopup.close();
+        card.removeCard();
+      })
+      .catch((err) => console.log(err))
+      .finally(() => deleteCardPopup.setSubmitText(false));
+  });
+}
+
 
+
+//FORM POPUP
 const openImagePopup = new PopupWithForm(
   "#add-new-card",
   handleImageFormSubmit
 );
-const editProfile = new PopupWithForm(
-  "#profile-edit-modal",
-  handleFormSubmit
+const editProfile = new PopupWithForm("#profile-edit-modal", handleFormSubmit);
+const profilePicEdit = new PopupWithForm(
+  "#profile__image-edit",
+  profilePicEditSubmit
 );
+
 editProfile.setEventListeners();
+profilePicEdit.setEventListeners();
 openImagePopup.setEventListeners();
 
-const editFormValidator = new FormValidator(config, profileModalForm);
-const cardFormValidator = new FormValidator(config, addNewCardModal);
+const handleEditFormSubmit = new FormValidator(config, profileModalForm);
+const handleProfileFormSubmit = new FormValidator(config, profilePicModal);
+const handleCardFormSubmit = new FormValidator(config, addNewCardModal);
 
-editFormValidator.enableValidation();
-cardFormValidator.enableValidation();
+handleEditFormSubmit.enableValidation();
+handleProfileFormSubmit.enableValidation();
+handleCardFormSubmit.enableValidation();
 
 //FORM SUBMIT
 function handleFormSubmit(data) {
-  console.log(data)
-  userInfo.setUserInfo(data)
-  editProfile.close();
+  editProfile.setSubmitText(true, "Saving...");
+  api
+    .profileUpdate(data)
+    .then(() => {
+      userInfo.setUserInfo(data);
+      editProfile.setSubmitText(false);
+      editProfile.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+
 }
 
 // //IMAGE SUBMIT
-function handleImageFormSubmit() {
-  const name = newCardModalTitle.value;
-  const link = newCardModalUrl.value;
-  generateCard({ name, link });
-  openImagePopup.close();
+function handleImageFormSubmit(data) {
+  const name = data.name;
+  const link = data.url;
+  openImagePopup.setSubmitText(true, "Saving...");
+  api
+    .addCard(name, link)
+    .then((card) => {
+      generateCard(card);
+      openImagePopup.setSubmitText(false)
+      openImagePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+  
+}
+
+function profilePicEditSubmit(data) {
+  const url = data.url;
+  profilePicEdit.setSubmitText(true, "Loading...");
+  api
+    .updateProfileAvatar(url)
+    .then(() => {
+      profileImage.src = url;
```

Comment:
`      profileImage.src = url;`

Class `UserInfo` should set the entire user’s data (including the **avatar**)

Reply:
I'm confused here. Are you saying I need to use the setUserInfo for just the picture?

Reply:
> I'm confused here. Are you saying I need to use the setUserInfo for just the picture?

You can make it work for one picture but... you need to use the **response** from the server. It contains all 3 params needed for `setUserInfo `